crates/sovereign-sdk/module-system/sov-modules-stf-blueprint/src/stf_blueprint.rs

```diff
@@ -63,15 +63,16 @@ where
     pub fn apply_sov_txs_inner(
         &mut self,
         soft_confirmation_info: HookSoftConfirmationInfo,
-        txs: Vec<Vec<u8>>,
+        txs: &[Vec<u8>],
         mut sc_workspace: WorkingSet<C>,
     ) -> (WorkingSet<C>, Vec<TransactionReceipt<TxEffect>>) {
         let mut tx_receipts = Vec::with_capacity(txs.len());
         for raw_tx in txs {
-            let raw_tx_hash = <C as Spec>::Hasher::digest(&raw_tx).into();
+            let raw_tx_hash = <C as Spec>::Hasher::digest(raw_tx).into();
             // Stateless verification of transaction, such as signature check
             // TODO: https://github.com/chainwayxyz/citrea/issues/1061
-            let tx = Transaction::<C>::deserialize_reader(&mut &*raw_tx)
+            let mut reader = std::io::Cursor::new(raw_tx);
```

Comment:
deserialize_reader requires a mutable reader, and I can not get mutable slice since it would require making tons of methods and structs mutable all the way up. So I have to create a slice reader.